lua/lspconfig/cmake.lua

```diff
@@ -1,11 +1,16 @@
 local configs = require 'lspconfig/configs'
 local util = require 'lspconfig/util'
 
+root_pattern = util.root_pattern(".git", "compile_commands.json", "build");
 configs.cmake = {
   default_config = {
     cmd = {"cmake-language-server"};
     filetypes = {"cmake"};
-    root_dir = util.root_pattern(".git", "compile_commands.json", "build");
+    root_dir = function(fname)
+      local filename = util.path.is_absolute(fname) and fname
+        or util.path.join(vim.loop.cwd(), fname)
+      return root_pattern(filename) or util.path.dirname(filename)
```

Comment:
This should be able to be just  `util.path.dirname(fname)`, no? The above two lines can be deleted.

Reply:
Also, I would keep root_pattern in the return instead of moving it to line 4 so our automated docgen can copy the function text into the docs.

Reply:
I don't know lua, I copy the configuration of clangd

Reply:
No worries :)  It's correct now